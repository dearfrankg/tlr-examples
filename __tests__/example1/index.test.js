import { render, screen } from "@testing-library/react";
import Home from "@/components/example1";

/*
 * EXAMPLE1: TEST snapshot and getByRole
 * SOURCE: nextjs example: `npx create-next-app --example with-jest <app_name>`
 */
describe("Example1: Home", () => {
  it("renders homepage unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  it("renders a heading", () => {
    render(<Home />);

    const heading = screen.getByRole("heading", {
      name: /Testing Library React Examples/i,
    });

    expect(heading).toBeInTheDocument();
  });
});
