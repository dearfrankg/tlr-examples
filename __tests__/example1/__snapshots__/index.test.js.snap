// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example1: Home renders homepage unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        Testing Library React Examples
      </h1>
      <pre>
        <code>
          cd app_dir; yarn test
        </code>
      </pre>
    </main>
  </div>
</div>
`;
